server/services/database/profiles.js

```diff
@@ -35,10 +35,15 @@ const createProfile = (newProfile) => {
 };
 
 const deleteProfile = (profileId) => {
-	return pool
-		.query("DELETE FROM profiles WHERE id = $1", [profileId])
-		.then((result) => result.rows);
-};
+	pool
+		.query("DELETE FROM profile_group WHERE profile_id= $1 ",[profileId]);
+
+	 if(profileId) {
+     return pool
+     .query("DELETE FROM profiles WHERE id = $1",[profileId]);
+	};
+}
+
```

Comment:
```suggestion
  return pool
    .query("DELETE FROM profile_group WHERE profile_id = $1", [profileId])
    .then(() => {
      return pool.query("DELETE FROM profiles WHERE id = $1", [profileId]);
    });

```
I think this should return the promise for the query as we do for the insert, chaining the database calls sequentially. The `if(profileId)` shoudln't be needed here either.

We're still not strictly doing things correctly since we are using `pool.query` repeatedly and not using any [transactions](https://node-postgres.com/features/transactions/) but this will do for now :slightly_smiling_face: .